server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MDNSendMethodContract.scala

```diff
@@ -0,0 +1,1948 @@
+                   |    "k1546": {
+                   |        "type": "notFound",
+                   |        "description": "The reference \"forEmailId\" cannot be found."
+                   |    }
+                   |}""".stripMargin)
+  }
+
+  @Test
+  def mdnSendShouldReturnNotFoundWhenMessageRelateHasNotDispositionNotificationTo(guiceJamesServer: GuiceJamesServer): Unit = {
+    val mailboxProbe: MailboxProbeImpl = guiceJamesServer.getProbe(classOf[MailboxProbeImpl])
+
+    val bobMailBoxPath: MailboxPath = MailboxPath.inbox(BOB)
+    mailboxProbe.createMailbox(bobMailBoxPath)
+
+    val andreMailBoxPath: MailboxPath = MailboxPath.inbox(ANDRE)
+    mailboxProbe.createMailbox(andreMailBoxPath)
+
+    val message: Message = Message.Builder
+      .of
+      .setSubject("test")
+      .setSender(BOB.asString)
+      .setFrom(BOB.asString)
+      .setTo(ANDRE.asString)
+      .setBody("Body of mail, that mdn related", StandardCharsets.UTF_8)
+      .build
+
+    val relatedEmailId: MessageId = mailboxProbe
+      .appendMessage(ANDRE.asString(), andreMailBoxPath, AppendCommand.builder()
+        .build(message))
+      .getMessageId
+
+    val request: String =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "urn:ietf:params:jmap:mail",
+         |    "urn:ietf:params:jmap:mdn"
+         |  ],
+         |  "methodCalls": [
+         |    [
+         |      "MDN/send",
+         |      {
+         |        "accountId": "$ANDRE_ACCOUNT_ID",
+         |        "identityId": "$ANDRE_IDENTITY_ID",
+         |        "send": {
+         |          "k1546": {
+         |            "forEmailId": "${relatedEmailId.serialize()}",
+         |            "disposition": {
+         |              "actionMode": "manual-action",
+         |              "sendingMode": "mdn-sent-manually",
+         |              "type": "displayed"
+         |            }
+         |          }
+         |        },
+         |        "onSuccessUpdateEmail": {
+         |          "#k1546": {
+         |            "keywords/mdnsent": true
+         |          }
+         |        }
+         |      },
+         |      "c1"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    val response: String =
+      `given`(
+        baseRequestSpecBuilder(guiceJamesServer)
+          .setAuth(authScheme(UserCredential(ANDRE, ANDRE_PASSWORD)))
+          .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+          .setBody(request)
+          .build, new ResponseSpecBuilder().build)
+        .post
+      .`then`
+        .statusCode(SC_OK)
+        .contentType(JSON)
+        .extract
+        .body
+        .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1].notSent")
+      .isEqualTo("""{
+                    |    "k1546": {
+                    |        "type": "notFound",
+                    |        "description": "Invalid \"Disposition-Notification-To\" header field."
```

Comment:
I believe a invalidArgument would be a better fit.

Reply:
But, The RFC has been defined:
```
notFound:
The reference "forEmailId" cannot be found or has no valid "Disposition-Notification-To" header field
```